app/navigation/OnshoreNavigator.scala

```diff
@@ -127,12 +128,13 @@ class OnshoreNavigatorImpl @Inject()(uaToDisclosure: UAToDisclosureService) exte
       val lettingsChosen = ua.get(WhatOnshoreLiabilitiesDoYouNeedToDisclosePage).getOrElse(Set()).contains(LettingIncome)
       val offshoreAnswer = ua.get(OffshoreLiabilitiesPage).getOrElse(true)
       val onshoreAnswer = ua.get(OnshoreLiabilitiesPage).getOrElse(true)
+      val properties = ua.get(LettingPropertyPage)
       val offshoreOnshoreBothSelected = (offshoreAnswer && onshoreAnswer)
 
       ua.get(WhichOnshoreYearsPage) match {
         case Some(years) if years.contains(PriorToFiveYears) && years.size == 1 && offshoreOnshoreBothSelected => routes.YouHaveNoOnshoreLiabilitiesToDiscloseController.onPageLoad
         case Some(years) if years.contains(PriorToFiveYears) && years.size == 1 => routes.MakingNilDisclosureController.onPageLoad
-        case Some(years) if lettingsChosen => routes.PropertyAddedController.onPageLoad(NormalMode)
+        case Some(years) if (lettingsChosen && properties.size == 0) => routes.PropertyAddedController.onPageLoad(NormalMode)
```

Comment:
You're checking the size on the Option here, rather than the list inside the Option. I think this is fine but we can just use "isEmpty" instead to make that clear

Reply:
Fixed.